head/src/test/kotlin/io/qalipsis/core/head/report/DataSeriesServiceImplTest.kt

```diff
@@ -287,6 +292,58 @@ internal class DataSeriesServiceImplTest {
             }
         }
 
+        @Test
+    internal fun `should not create the data series with a color opacity above 100`() =
+        testDispatcherProvider.runTest {
+            // given
+            coEvery { dataSeriesRepository.save(any()) } returnsArgument 0
+            coEvery { tenantRepository.findIdByReference("my-tenant") } returns 123L
+            coEvery { userRepository.findIdByUsername("my-user") } returns 456L
+            coEvery { idGenerator.short() } returns "the-reference"
+            val dataSeries = DataSeriesCreationRequest(
+                displayName = "the-name",
+                dataType = DataType.EVENTS,
+                color = "#ff761c",
+                filters = setOf(DataSeriesFilter("field-1", QueryClauseOperator.IS_IN, "A,B")),
+                fieldName = null,
+                aggregationOperation = null,
+                timeframeUnit = Duration.ofSeconds(2),
+                displayFormat = "#0.000",
+                colorOpacity = 101
+            )
+
+            // when
+            assertThrows<IllegalArgumentException> {
```

Comment:
I don't think that an exception can be thrown here, because you do not user the Micronaut runtime that triggers the validation around the calls. Such a test can only occur when using either the controller or the repository.